### PR TITLE
Adding changes for add_application-need to add db entry to test(AM_SUBSCRIBER)

### PR DESCRIPTION
**Pull request by thushara35**
Need to fix below jiras from ES side to work smoothly 
https://wso2.org/jira/browse/STORE-692
https://wso2.org/jira/browse/STORE-693

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 1]
```diff
@@ -1960,15 +1964,16 @@ public String addApplication(String appName, String userName, String tier, Strin
 			throws APIManagementException {
 
 		Subscriber subscriber = new Subscriber(username);
-		JSONArray apps = getApplications(username);
+        Application[] apps = getAllApplications(username);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Slight formatting issue in here!

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 2]
```diff
@@ -1960,15 +1964,16 @@ public String addApplication(String appName, String userName, String tier, Strin
...
 
-		for (int i = 0; i < apps.size(); i++) {
-			Application app = (Application) apps.get(i);
-			if (app.getName().equals(appName)) {
-				handleException("A duplicate application already exists by the name - "
-						+ appName);
-			}
-		}
+        if (apps != null) {
+            for (int i = 0; i < apps.length; i++) {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: How about using
```java
for(Application app : apps) {
                if (app.getName().equals(appName)) {
                    handleException("A duplicate application already exists by the name - " + appName);
                }
}
java
                }
}

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/extensions/store/app/am_store/app.js [hunk 3]
```diff
@@ -24,6 +24,10 @@ app.server = function(ctx) {
                 url: 'widget',
                 path: 'widget.jag',
                 secured:false
+            }, {
+                url: 'applications',
+                path: 'applications.jag',
+                secured:false
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/extensions/store/app/am_store/app.js]: what will be the impact of setting secured attribute to false. Will it make users to access this page without authenticating

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 1 above)
Comment (thushara35) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: this is can not be replicated locally

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 2 above)
Comment (thushara35) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Thanks for the input

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/extensions/store/app/am_store/app.js (hunk 3 above)
Comment (thushara35) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/extensions/store/app/am_store/app.js]: boolean should be true. This code indicates whether a user should be authenticated prior to been allowed to access the Page or API.If the property is not specified then the endpoint is assumed to be publicly accessible.

**Pull request merged**